RIOT-OS/RIOT
# sys: Basic C++ compatibility with C11 atomcis

### Contribution description

- Added `c11_atomics_compat.hpp` to `sys` to provide types with the same name and memory layout as the default C11 atomic types (`atomic_bool`, `atomic_int`, `atomic_uint_fast16_t`, ...)
    - As size and memory layout depend on the CPU being used, a platform specific header needs to be provided per platform. (The same file can be used for all ARM boards, so not as bad as it sounds.)
- Added a helper (shell) script to generate the platform specific header
    - A working C compiler and ~3 seconds of time are needed to generate the header
- Updated `core/rmutex` as proof of concept
- Added `tests/rmutex_cpp` which is an unmodified copy of`test/rmutex`, but:
     - `main.c` was renamed to `main.cpp` to get it compiled as C++ code
     - Added `cpp` as required feature

### Testing procedure

- `tests/rmutex_cpp` will not compile with master (when cherry-picked on top), but should compile and run properly with this PR
- `tests/rmutex` should still compile and run properly

### Issues/PRs references

Depends on and includes https://github.com/RIOT-OS/RIOT/pull/12566.

For the SPI Arduino compatibility wrapper in https://github.com/RIOT-OS/RIOT/pull/12118 rmutex is needed to provide full compatibility. But due to the inherent incompatibility between C11 atomics and C++, using rmutex was not possible yet. This PR fixes this.
